server/src/test/java/io/crate/auth/AuthenticationIntegrationTest.java

```diff
@@ -75,16 +76,16 @@ protected Settings nodeSettings(int nodeOrdinal) {
                 "d", new String[]{"user", "method", "address"}, new String[]{"arthur", "trust", "127.0.0.1"})
             .put("auth.host_based.config",
                 "e",
-                new String[]{"user", HostBasedAuthentication.SSL.KEY},
-                new String[]{"requiredssluser", HostBasedAuthentication.SSL.REQUIRED.VALUE})
+                new String[]{"user", "ssl"},
+                new String[]{"requiredssluser", SslSettings.SSLMode.ON.name()})
             .put("auth.host_based.config",
                 "f",
-                new String[]{"user", HostBasedAuthentication.SSL.KEY},
-                new String[]{"optionalssluser", HostBasedAuthentication.SSL.OPTIONAL.VALUE})
```

Comment:
This would be a breaking change -  the setting value that was previously `optional` would now be `dual`. We can't do that.

We'll need to stick with the two enums and fix the equality checks where necessary

Reply:
I see, good spot 👍 

Regarding equality checks  - I expect don't use simple hack like ... || "true" cause it's not future proof  - but rather use SSL enum and provide parse method to handle `on => true` issue, so that it doesn't pop up in another place - and it would be almost copy paste of the new enum.

What if we handle optional in `SSLMode.parse` just like we handle `on/true` to avoid such mix of 2 enums? Or, as "dual" is not yet released, rename it to the optional?

Or, If I stick with 2 enums is it ok to have dedicated parse in SSL similar to SSLMode one?

Reply:
Why not just something like this?

```
diff --git a/server/src/main/java/io/crate/auth/HostBasedAuthentication.java b/server/src/main/java/io/crate/auth/HostBasedAuthentication.java
index 2e6b11928ea..7b1d746ba32 100644
--- a/server/src/main/java/io/crate/auth/HostBasedAuthentication.java
+++ b/server/src/main/java/io/crate/auth/HostBasedAuthentication.java
@@ -196,15 +196,17 @@ static boolean isValidAddress(@Nullable String hbaAddressOrHostname, InetAddress
         static boolean isValidProtocol(String hbaProtocol, Protocol protocol) {
             return hbaProtocol == null || hbaProtocol.equals(protocol.toString());
         }

         static boolean isValidConnection(String hbaConnectionMode, ConnectionProperties connectionProperties) {
-            return hbaConnectionMode == null ||
-                   hbaConnectionMode.isEmpty() ||
-                   hbaConnectionMode.equals(SSL.OPTIONAL.VALUE) ||
-                   (hbaConnectionMode.equals(SSL.NEVER.VALUE) && !connectionProperties.hasSSL()) ||
-                   (hbaConnectionMode.equals(SSL.REQUIRED.VALUE) && connectionProperties.hasSSL());
+            if (hbaConnectionMode == null || hbaConnectionMode.isEmpty()) {
+                return true;
+            }
+            SSL sslMode = SSL.valueOf(hbaConnectionMode);
+            switch (sslMode) {
+                case ...
+            }
         }

         private static long inetAddressToInt(InetAddress address) {
             long net = 0;
             for (byte a : address.getAddress()) {
```

The new enum that was introduced with `dual` is used to control the servers transport SSL mode, it has no relation to the HBA setting.

If I understand the problem correctly, this may even be a bug that is unrelated to the new node-to-node encryption feature?

Reply:
I used code above but used a new parseValue method - I cannot directly use valueOf as "true" and "false" are special cases. I see your point about completely different config and need of another enum, makes sense.

> unreleated

it's not part of new feature but it affects node-to-node in some cases - when I want to set `ssl: on` like in example https://crate.io/docs/crate/reference/en/master/admin/auth/hba.html#node-to-node-communication